Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -1321,83 +1321,246 @@ auto RenderTheme::colorCache(OptionSet<StyleColorOptions> options) const -> Colo
 Color RenderTheme::systemColor(CSSValueID cssValueId, OptionSet<StyleColorOptions> options) const
 {
     switch (cssValueId) {
-    case CSSValueWebkitLink:
-        return options.contains(StyleColorOptions::ForVisitedLink) ? SRGBA<uint8_t> { 85, 26, 139 } : SRGBA<uint8_t> { 0, 0, 238 };
-    case CSSValueWebkitActivelink:
-    case CSSValueActivetext:
-        return Color::red;
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-canvas
+    // Background of application content or documents.
+    case CSSValueCanvas:
+        return Color::white;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-canvastext
+    // Text in application content or documents.
+    case CSSValueCanvastext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-linktext
+    // Text in non-active, non-visited links. For light backgrounds, traditionally blue.
     case CSSValueLinktext:
         return SRGBA<uint8_t> { 0, 0, 238 };
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-visitedtext
+    // Text in visited links. For light backgrounds, traditionally purple.
     case CSSValueVisitedtext:
         return SRGBA<uint8_t> { 85, 26, 139 };
-    case CSSValueActiveborder:
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-activetext
+    // Text in active links. For light backgrounds, traditionally red.
+    case CSSValueActivetext:
+    case CSSValueWebkitActivelink: // Non-standard addition.
+        return Color::red;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttonface
+    // The face background color for push buttons.
+    case CSSValueButtonface:
+        return Color::lightGray;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttontext
+    // Text on push buttons.
+    case CSSValueButtontext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttonborder
+    // The base border color for push buttons.
+    case CSSValueButtonborder:
+        return Color::white;
```

Comment:
These reflect the view that "you can make a standard button using CSS", which obviously seems untrue for the style of buttons on Apple platforms. I suppose then the color values are arbitrary since these can’t be used to make buttons that conform to the platform standards for what buttons look like?

Reply:
I do really question why these exist (and are being added to) these days, as this concept really harkens back to a very Windows centric web (most of the existing names are straight out of the Win32 SDK). I really don't know what these could be used for in practice, as there just aren't enough of them to fully express what platform buttons look like (on macOS I am pretty sure <button> has shadow/gradient that defines the border for instance). 

There are some WPT tests that seem to suggest there is some expectation of these values matching UA stylesheet values, but I can't find any specification for that, and WebKit has really never had this property so I doubt it is required for compat.  

I filed https://github.com/w3c/csswg-drafts/issues/7781 to ask the editors about it.

On the other hand, we already support so many of these, that adding these final few hardly seems like it will tip the scales in any direction.